packages/x-date-pickers/src/internals/hooks/date-helpers-hooks.tsx

```diff
@@ -49,7 +49,8 @@ export function useMeridiemMode<TDate>(
 
   const handleMeridiemChange = React.useCallback(
     (mode: 'am' | 'pm') => {
-      const timeWithMeridiem = convertToMeridiem<TDate>(date, mode, Boolean(ampm), utils);
+      const timeWithMeridiem =
```

Comment:
I think the methods in `date-io` actually handles the nullish value correctly most of the time
But there typing says otherwise, so I prefer to play safe here.
The date can be null, the `getHours` method used inside `convertToMeridiem` do not accept `null` so I don't call it when the date is null.